pusher-platform-core/src/main/kotlin/com/pusher/platform/subscription/BaseSubscription.kt

```diff
@@ -82,36 +88,32 @@ internal class BaseSubscription(
     }
 
     private fun handleConnectionFailed(response: Response) {
-        if (response.body() != null) {
-            val errorEvent = response.body()?.charStream()
-                .parseOr { ErrorResponseBody("Could not parse: $response") }
-                .fold(
-                    onFailure = { it },
-                    onSuccess = {
-                        ErrorResponse(
-                            statusCode = response.code(),
-                            headers = response.headers().toMultimap(),
-                            error = it.error,
-                            errorDescription = it.errorDescription,
-                            URI = it.URI
-                        )
-                    }
-                )
-            onError(errorEvent)
-        }
+        val errorEvent = response.body()?.charStream()
+            .parseOr { ErrorResponseBody("Could not parse: $response") }
+            .map {
+                ErrorResponse(
+                    statusCode = response.code(),
+                    headers = response.headers().toMultimap(),
+                    error = it.error,
+                    errorDescription = it.errorDescription,
+                    URI = it.URI
+                ) as Error
+            }
+            .flatten()
+        onError(errorEvent)
     }
 
     private fun handleConnectionOpened(response: Response) {
         onOpen(response.headers().toMultimap())
 
         val body = response.body()
-
+        activeResponseBody = body
```

Comment:
What's `activeBodyResponse` for? I can't differentiate it from `body`

Reply:
This one is to close the body when the subscription is closed